frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/EmbedModalContent.tsx

```diff
@@ -0,0 +1,234 @@
+import { useState } from "react";
+import _ from "underscore";
+import { getSignedPreviewUrl, getSignedToken } from "metabase/public/lib/embed";
+import { getSetting } from "metabase/selectors/settings";
+import { useSelector } from "metabase/lib/redux";
+import type { ExportFormatType } from "metabase/dashboard/components/PublicLinkPopover/types";
+
+import { StaticEmbedSetupPane } from "../StaticEmbedSetupPane";
+import type {
+  ActivePreviewPane,
+  EmbeddingDisplayOptions,
+  EmbeddingParameters,
+  EmbeddingParametersValues,
+  EmbedResource,
+  EmbedResourceParameter,
+  EmbedResourceType,
+  EmbedType,
+} from "../EmbedModal.types";
+import { SelectEmbedTypePane } from "../SelectEmbedTypePane";
+import { EmbedModalContentStatusBar } from "./EmbedModalContentStatusBar";
+
+const FALLBACK_SECRET_KEY = "<METABASE_SECRET_KEY>";
+
+export interface EmbedModalContentProps {
+  embedType: EmbedType;
+  setEmbedType: (type: EmbedType) => void;
+
+  resource: EmbedResource;
+  resourceType: EmbedResourceType;
+  resourceParameters: EmbedResourceParameter[];
+
+  onUpdateEnableEmbedding: (enableEmbedding: boolean) => void;
+  onUpdateEmbeddingParams: (embeddingParams: EmbeddingParameters) => void;
+
+  onCreatePublicLink: () => void;
+  onDeletePublicLink: () => void;
+  getPublicUrl: (
+    resource: EmbedResource,
+    extension?: ExportFormatType,
+  ) => string | null;
+
+  className?: string;
+}
+
+export const EmbedModalContent = (
+  props: EmbedModalContentProps,
+): JSX.Element => {
+  const {
+    embedType,
+    setEmbedType,
+    resource,
+    resourceType,
+    resourceParameters,
+    onUpdateEnableEmbedding,
+    onUpdateEmbeddingParams,
+    onCreatePublicLink,
+    onDeletePublicLink,
+    getPublicUrl,
+  } = props;
+
+  const [pane, setPane] = useState<ActivePreviewPane>("code");
+
+  const siteUrl = useSelector(state => getSetting(state, "site-url"));
+  const secretKey =
+    useSelector(state => getSetting(state, "embedding-secret-key")) ||
+    // "secretKey" should always exist if embedding has been turned on, but to fix typings we add here a fallback value
+    FALLBACK_SECRET_KEY;
+
+  const [embeddingParams, setEmbeddingParams] = useState<EmbeddingParameters>(
+    getDefaultEmbeddingParams(resource, resourceParameters),
+  );
+  const [parameterValues, setParameterValues] =
+    useState<EmbeddingParametersValues>({});
+  const [displayOptions, setDisplayOptions] = useState<EmbeddingDisplayOptions>(
+    {
+      font: null,
+      theme: null,
+      bordered: true,
+      titled: true,
+    },
+  );
+
+  const handleSave = async () => {
+    try {
+      if (embedType === "application") {
```

Comment:
I don't think we will have another `embedType` than `"application"` in this function because we only pass `handleSave` after we have checked that the `embedType` couldn't be null. And `embedType` can only either be `null` or `"application"`.

I suspect the else block was from before we passed `onCreatePublicLink` directly to `SelectEmbedTypePane`, this really emphasizes my other comment about splitting the component for each modal, so it's easier to reason about and handle their different logics.